src/config/api/utils.ts

```diff
@@ -0,0 +1,22 @@
+import api from '.';
+
+const baseHeaders = ['Accept', 'Content-Type'];
+
+const removeHeaders = () => {
+  Object.keys(api.headers).forEach(header => {
+    if (!baseHeaders.includes(header)) {
+      api.deleteHeader(header);
+    }
+  });
+};
+
+const setHeaders = (headers: Record<string, string>) => {
```

Comment:
the problem here is that you are exposing a function that `api.setHeaders` already does, it is better to make the specific logic here, typed exactly as what you expect, with the specific properties, and make the set explicit.